analyzers/tests/SonarAnalyzer.TestFramework/Verification/Verifier.cs

```diff
@@ -97,15 +97,19 @@ public Verifier(VerifierBuilder builder)
 
     public void Verify()    // This should never have any arguments
     {
-        if (codeFix != null)
+        if (codeFix is not null)
         {
             throw new InvalidOperationException($"Cannot use {nameof(Verify)} with {nameof(builder.CodeFix)} set.");
         }
-        foreach (var compilation in Compile(builder.ConcurrentAnalysis))
-        {
-            var additionalSourceFiles = razorFilePaths.Concat(compilation.AdditionalSourceFiles ?? []).ToArray();
-            DiagnosticVerifier.Verify(compilation.Compilation, analyzers, builder.ErrorBehavior, builder.AdditionalFilePath, onlyDiagnosticIds, additionalSourceFiles);
-        }
+        var numberOfIssues = Compile(builder.ConcurrentAnalysis)
+            .Sum(x => DiagnosticVerifier.Verify(
+                x.Compilation,
+                analyzers,
+                builder.ErrorBehavior,
+                builder.AdditionalFilePath,
+                onlyDiagnosticIds,
+                razorFilePaths.Concat(x.AdditionalSourceFiles ?? []).ToArray()));
+        numberOfIssues.Should().BeGreaterThan(0, $"otherwise you should use '{nameof(VerifyNoIssues)}' instead");
```

Comment:
The suggestion came up from FluentAssertion analyzers as well and I marked it as Accepted. I find `BeGreaterThan(0)` far more clear than `BePositive()`. Whenever I find the latter, I'm wondering whether or not it includes zero.

Reply:
The message is written in a way that it fits into the error message created by the assertion, which is something like "'numberOfIssues' should be greater than zero, because <because clause>, but was 0 instead.".